### PR TITLE
make make sure e2e comes after build

### PR DESCRIPTION
**Pull request by virgilchiriac**
# Description

moves e2e with each commit, to have them testing each image build
makes sure image exists, before the e2e runs

## Links to Tickets or other pull requests

https://github.com/hpi-schul-cloud/schulcloud-server/pull/3000
https://github.com/hpi-schul-cloud/nuxt-client/pull/1904
- https://ticketsystem.schul-cloud.org/browse/BC-68

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: .github/workflows/build_and_e2e.yml [hunk 1]
```diff
@@ -1,10 +1,9 @@
 ---
-name: Build and push Docker Image
+name: Build Image and E2E
```
Comment (mamutmk5) [inline .github/workflows/build_and_e2e.yml:2]: Better name where Push workflow, also for file push.yml

**Review by mamutmk5 — CHANGES_REQUESTED** [review]
I suggest for the file the name push.yml and as workflow name: push workflow

File: .github/workflows/build_and_e2e.yml (hunk 1 above)
Comment (virgilchiriac) [reply .github/workflows/build_and_e2e.yml]: done

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -1,10 +1,9 @@
 ---
-name: Build and push Docker Image
+name: push workflow
 
 on:
   push:
     branches-ignore:
-      - main
       - master
```
Comment (mamutmk5) [inline .github/workflows/push.yml:7]: why have you not remove master?

**Review by mamutmk5 — CHANGES_REQUESTED** [review]

File: .github/workflows/push.yml (hunk 2 above)
Comment (virgilchiriac) [reply .github/workflows/push.yml]: done

**Review by mamutmk5 — APPROVED** [review]

**Pull request merged**